runelite-client/src/main/java/net/runelite/client/util/SwingUtil.java

```diff
@@ -207,11 +213,22 @@ public void windowClosing(WindowEvent event)
 				{
 					if (confirmRequired.call())
 					{
+						JPanel content = new JPanel();
+						content.setLayout(new GridLayout(2, 1, 0, 10));
+
+						JLabel text = new JLabel("Are you sure you want to exit?");
+						content.add(text, 0);
+
+						JCheckBox checkBox = new JCheckBox("Never ask again");
+						content.add(checkBox, 1);
+
 						result = JOptionPane.showConfirmDialog(
 							frame,
-							"Are you sure you want to exit?", "Exit",
+								content, "Exit",
 							JOptionPane .OK_CANCEL_OPTION,
 							JOptionPane.QUESTION_MESSAGE);
+
+						config.setConfirmOnClose(!checkBox.isSelected());
```

Comment:
Now you changed this method from fully independent on how rest of the application works to fully dependant. I don't think this call to config should be here.

Reply:
Could you suggest an alternative?
I could remove the checkbox(and thus the call to config) from the dialog itself and just have the option in RuneLiteConfig?